### DISCUSSION TITLE
Interest in Contributing to HPX for GSoC 2025

### DISCUSSION THREAD

**Discussion by NishantBansal2003** [STEllAR-GROUP/hpx#6554] [General]
Hi hpx team,

I’m Nishant Bansal, an undergraduate student pursuing a Bachelor of Technology at IIT (BHU), India. I have experience in C++(particularly in Data Structures and Algorithms), and I’m enthusiastic about contributing to open-source projects. I’m particularly interested in hpx and would love to participate in Google Summer of Code (GSoC) 2025 with your team.

As I begin exploring hpx, I would greatly appreciate your guidance on the following:

- **Project Scope:** Are there any specific areas or components within hpx that are prioritised for GSoC 2025? I’m eager to contribute to projects that align with my experience in C++.

- **Getting Started:** What is the best way to get familiar with the hpx codebase? Are there beginner-friendly issues or tasks that would be suitable for someone starting to explore hpx repo?

- **Testing Changes Locally:** Could you provide guidance on how to run hpx locally? What is the recommended approach to set up the project, build it, and test changes? Additionally, are there specific testing tools or frameworks used in hpx? Any tips on how to validate that my changes don’t break anything in the codebase would be very helpful.

- **Preparation:** Are there any particular tools, technologies, or concepts I should focus on to contribute effectively to the hpx project?

I’m excited about the possibility of contributing and learning through this project, and I look forward to your guidance!

Best regards,
Nishant Bansal

**Comment by Pansysk75** [comment]
Hello Nishant,
Thank you for your interest in contributing to HPX, let me try to answer your questions:

We still haven't prepared our GSoC page for 2025, however, you can refer to last year's GSoC wiki for past project ideas and tips on getting started with HPX:
https://github.com/STEllAR-GROUP/hpx/wiki/Google-Summer-of-Code-%28GSoC%29-2024#tips-for-prospective-students

Please try the steps outlined, and feel free to ask questions or start discussions here or on our Discord.

For building and testing, HPX uses CMake and CTest. There is a lot of information about building HPX locally on the documentation page, but feel free to ask questions if anything is unclear:
https://hpx-docs.stellar-group.org/latest/html/manual/building_hpx.html
